### ISSUE TITLE
"Split vector Layer" error with GPKG inputs

### ISSUE THREAD

**Issue by Fi156** [qgis/QGIS#36378]
[aaa.gpkg.zip](https://github.com/qgis/QGIS/files/5735108/aaa.gpkg.zip)
Do the following:
- Create new GPKG and Layer
- Add some Points
- Add two new Field: fid_2 and SplitID
- Fill in your data for each row:
fid_2 has to be unique
SplitID repeat number 1 and 2 for all your points.
- Save layer
- Right click --> Export as... Layer
- In Layer Options, **set FID = fid_2**
- Save layer as tmp2 or something.
- Open that layer and edit all fid values to one number (eg all to "1")

Try to use: "Split vector Layer" with "Unique Field ID" = SplitID

Error: 
First feature for every new split ID will be written. All others not:
"Feature could not be written to <Path to File>"

As soon as you delete the "fid" field, the tool will not give the error and work as exspected.

Long story short:
When giving the FID a different name then "fid" and leave the original "fid" field with non unique data in, an error will occur.

(Sidenode: Very funny in conjunction with tools like "Points along geometry", which are not updating the original fid field)

Version: 3.10.3
[aaa.gpkg.zip](https://github.com/qgis/QGIS/files/5735110/aaa.gpkg.zip)

**Comment by gioman** [comment]
>     * Create new GPKG and Layer

@Fi156 it seems you have already created a sample dataset that shows the issue, can you attach it here please? thanks.

**Comment by jdlom** [comment]
I think it's not specific to the split vector layer algorithm.
The problem is about of the fid  field which is not unique in your dataset.

I try to export the dataset to another gpkg and I got the same error. 

You may consider a feature request  for the issue :
" export a layer with fid column which is not unique to gpkg in processing".

For the moment it is impossible  to give the primary key fields as creation options when choosing gpkg as default format.

**Comment by Pedro-Murteira** [comment]
@Fi156 Hello, is this still valid on more recent releases?
